network-grpc/src/client.rs

```diff
@@ -385,13 +385,19 @@ where
 
 impl<T, S, E> HeaderService<T> for Client<S, E>
 where
-    T: HasHeader,
+    T: Block + HasHeader,
```

Comment:
this is not necessarily true (that a block has a header).

Why not parameterising the trait with a `H: Header` parameter ?

Reply:
Fixed, it appeared to be a redundant change, left after previous work.